docs/examples/forecasting/extended_tutorial.md

```diff
@@ -0,0 +1,1746 @@
+                          custom_ds_metadata['num_steps'],                                                      
+                          custom_ds_metadata['prediction_length']
+                         )
+
+target, feat_dynamic_real, feat_static_cat = data_out
+```
+
+We can easily create the train and test datasets by simply filling in the correct fields. Remember that for the train dataset we need to cut the last window.
+
+
+```python
+train_ds = ListDataset([{FieldName.TARGET: target, 
+                         FieldName.START: start,
+                         FieldName.FEAT_DYNAMIC_REAL: fdr,
+                         FieldName.FEAT_STATIC_CAT: fsc} 
+                        for (target, start, fdr, fsc) in zip(target[:, :-custom_ds_metadata['prediction_length']], 
+                                                             custom_ds_metadata['start'], 
+                                                             feat_dynamic_real[:, :-custom_ds_metadata['prediction_length']], 
+                                                             feat_static_cat)],
+                      freq=custom_ds_metadata['freq'])
+```
+
+
+```python
+test_ds = ListDataset([{FieldName.TARGET: target, 
+                        FieldName.START: start,
+                        FieldName.FEAT_DYNAMIC_REAL: fdr,
+                        FieldName.FEAT_STATIC_CAT: fsc} 
+                       for (target, start, fdr, fsc) in zip(target, 
+                                                            custom_ds_metadata['start'], 
+                                                            feat_dynamic_real, 
+                                                            feat_static_cat)],
+                     freq=custom_ds_metadata['freq'])
+```
+
+Now, we can examine each entry of the train and test datasets. We should expect that they have the following fields: `target`, `start`, `feat_dynamic_real` and `feat_static_cat`.
+
+
+```python
+train_entry = next(iter(train_ds))
+train_entry.keys()
+```
+
+
+```python
+test_entry = next(iter(test_ds))
+test_entry.keys()
+```
+
+
+```python
+test_series = to_pandas(test_entry)
+train_series = to_pandas(train_entry)
+
+fig, ax = plt.subplots(2, 1, sharex=True, sharey=True, figsize=(10, 7))
+
+train_series.plot(ax=ax[0])
+ax[0].grid(which="both")
+ax[0].legend(["train series"], loc="upper left")
+
+test_series.plot(ax=ax[1])
+ax[1].axvline(train_series.index[-1], color='r') # end of train dataset
+ax[1].grid(which="both")
+ax[1].legend(["test series", "end of train series"], loc="upper left")
+
+plt.show()
+```
+
+<span style="color:red">*For the rest of the tutorial we will use the custom dataset*</span>
+
+
+```python
+
+```
+
+# 2. Transformation
+
+## 2.1 Define a transformation
+
+The job of a `Transformation` is to get an iterable collection of entries of a dataset and transform it to another iterable collection that contains possibly more fields useful for our forecasting application.
```

Comment:
can you add here what we would use a transformation for? something like this:

The primary use case for a transformation is for feature processing such as adding a holiday feature.

Reply:
What you're writing above is correct but may be a little to abstract.